lib/optimize/ModuleConcatenationPlugin.js

```diff
@@ -281,9 +281,12 @@ class ModuleConcatenationPlugin {
 							for (const reason of newModule.reasons) {
 								reason.dependency.module = newModule;
 							}
-							for (const dep of newModule.dependencies) {
+							for (let i = 0; i < newModule.dependencies.length; i++) {
```

Comment:
Could you add a comment explaining why this isn't a for-of loop pointing to this PR and adding `// TODO: remove in webpack 5`